lib/matplotlib/ticker.py

```diff
@@ -2502,64 +2654,119 @@ def view_limits(self, vmin, vmax):
+        """
+
+        self._minor = minor
+        MaxNLocator.__init__(self, nbins=nbins, steps=[1, 2, 5, 10])
+
+    @cbook._make_keyword_only("3.2", "minor")
+    def set_params(self, minor=None, **kwargs):
         """Set parameters within this locator."""
         if minor is not None:
-            self.minor = minor
+            self._minor = minor
+        MaxNLocator.set_params(self, **kwargs)
 
-    def __call__(self):
-        """Return the locations of the ticks."""
-        vmin, vmax = self.axis.get_view_interval()
-        return self.tick_values(vmin, vmax)
+    @property
+    def minor(self):
+        return self._minor
+
+    @minor.setter
+    def minor(self, value):
+        self.set_params(minor=value)
 
     def tick_values(self, vmin, vmax):
         # dummy axis has no axes attribute
-        if hasattr(self.axis, 'axes') and self.axis.axes.name == 'polar':
-            raise NotImplementedError('Polar axis cannot be logit scaled yet')
-
-        vmin, vmax = self.nonsingular(vmin, vmax)
-        vmin = np.log10(vmin / (1 - vmin))
-        vmax = np.log10(vmax / (1 - vmax))
+        if hasattr(self.axis, "axes") and self.axis.axes.name == "polar":
+            raise NotImplementedError("Polar axis cannot be logit scaled yet")
 
-        decade_min = np.floor(vmin)
-        decade_max = np.ceil(vmax)
-
-        # major ticks
-        if not self.minor:
-            ticklocs = []
-            if decade_min <= -1:
-                expo = np.arange(decade_min, min(0, decade_max + 1))
-                ticklocs.extend(10**expo)
-            if decade_min <= 0 <= decade_max:
-                ticklocs.append(0.5)
-            if decade_max >= 1:
-                expo = -np.arange(max(1, decade_min), decade_max + 1)
-                ticklocs.extend(1 - 10**expo)
-
-        # minor ticks
+        if self._nbins == "auto":
+            if self.axis is not None:
+                nbins = self.axis.get_tick_space()
+                if nbins < 2:
+                    nbins = 2
+            else:
+                nbins = 9
         else:
-            ticklocs = []
-            if decade_min <= -2:
-                expo = np.arange(decade_min, min(-1, decade_max))
-                newticks = np.outer(np.arange(2, 10), 10**expo).ravel()
-                ticklocs.extend(newticks)
-            if decade_min <= 0 <= decade_max:
-                ticklocs.extend([0.2, 0.3, 0.4, 0.6, 0.7, 0.8])
-            if decade_max >= 2:
-                expo = -np.arange(max(2, decade_min), decade_max + 1)
-                newticks = 1 - np.outer(np.arange(2, 10), 10**expo).ravel()
-                ticklocs.extend(newticks)
+            nbins = self._nbins
 
-        return self.raise_if_exceeds(np.array(ticklocs))
+        # We define ideal ticks with their index:
+        # linscale: ... 1e-3 1e-2 1e-1 1/2 1-1e-1 1-1e-2 1-1e-3 ...
+        # b-scale : ... -3   -2   -1   0   1      2      3      ...
+        def ideal_ticks(x):
+            return (
```

Comment:
I think the whole expression (without parentheses) fits in a single line? (I don't mind breaking it e.g. after each `else` for clarity, but here the linebreaks achieve little.)

Reply:
You were right. Originally I wrote this expression as a lambda fun, but my linter was complaining. Fixed.